File: _java-examples/BankAccount.java
public class BankAccount {
	
	private int balance;
	private int numberOfWithdrawals;


	public BankAccount(int amount){
		balance = amount;
	}

	public void deposit(int amount){
		if (amount>0)
			System.out.println("I am easily reachable in deposit");
			balance = balance + amount;
	}

	public void withdraw(int amount){
		if(amount>balance){
			System.out.println("I am easily reachable in withdraw");
			return;
		}
		if (numberOfWithdrawals>=5){// was 10
			assert(false);
			System.out.println("I am very hard to reach in withdraw");
			return;
		}
		balance = balance - amount;
		numberOfWithdrawals++;
	}

	/* ----- Test Driver ----- */

	public static void testDriver(int length){
		BankAccount b = new BankAccount(0);
		for (int i=0; i<length; i++){
			switch(flag(true)) {
				case 0:
					b.deposit(10);
					break;
				case 1:
					b.withdraw(1);
					break;
			}
		}
	}

	public static int flag(boolean x) {
	if (x)
		return 1;
	else
		return 0;
	}

	public static void main(String[] args){
		testDriver(3);
		System.out.println();
	}
}

File: _python-examples/bank_account.py
class BankAccount:

    balance = 0
    num_of_withdrawals = 0

    def __init__(self, amount: int):
        self.balance = amount

    def deposit(self, amount: int):
        if amount > 0:
            print("I am easily reachable in deposit")
            self.balance = self.balance + amount

    def withdraw(self, amount: int):
        if amount > self.balance:
            print("I am easily reachable in withdraw")
            return
        if self.num_of_withdrawals >= 5:
            assert False
            print("I am very hard to reach in withdraw")
            return
        self.balance = self.balance - amount
        self.num_of_withdrawals += 1


# Test driver
def flag(x: bool):
    if x:
        return 1
    else:
        return 0


def test_driver(length: int):
    b = BankAccount(0)
    for i in range(length):
        if flag(True) == 0:
            b.deposit(10)
        else:
            b.withdraw(1)


if __name__ == '__main__':
    test_driver(3)
